### PR TITLE
Enable local attention during decode

### PR DESCRIPTION
**Pull request by CatherineSue**
## Motivation
Support local (chunked) attention in decode phase

## Modifications
- Introduce _init_local_attn_metadata as a single utility method to handle initialization of local_attn_metadata across all relevant code.
- Add support for local (chunked) attention virtual batches in the decode phase in FlashAttentionBackend.
- Add local_attn_metadata initialization to `init_forward_metadata_replay_cuda_graph`.
- Refactor attention chunk size normalization logic into `make_local_attention_virtual_batches` to avoid code duplication.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [x] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Comment by CatherineSue** [comment]
Llama-4-Scout:

gsm8k:
```
Accuracy: 0.935
Invalid: 0.000
Latency: 20.778 s
Output throughput: 944.739 token/s
```

mmlu_pro:
```
|       Tasks       |Version|    Filter    |n-shot|  Metric   |   |Value |   |Stderr|
|-------------------|------:|--------------|-----:|-----------|---|-----:|---|-----:|
|mmlu_pro           |      2|custom-extract|      |exact_match|↑  |0.7513|±  |0.0038|
| - biology         |      1|custom-extract|     0|exact_match|↑  |0.8759|±  |0.0123|
| - business        |      1|custom-extract|     0|exact_match|↑  |0.7871|±  |0.0146|
| - chemistry       |      1|custom-extract|     0|exact_match|↑  |0.8074|±  |0.0117|
| - computer_science|      1|custom-extract|     0|exact_match|↑  |0.7610|±  |0.0211|
| - economics       |      1|custom-extract|     0|exact_match|↑  |0.8282|±  |0.0130|
| - engineering     |      1|custom-extract|     0|exact_match|↑  |0.6522|±  |0.0153|
| - health          |      1|custom-extract|     0|exact_match|↑  |0.7421|±  |0.0153|
| - history         |      1|custom-extract|     0|exact_match|↑  |0.6352|±  |0.0247|
| - law             |      1|custom-extract|     0|exact_match|↑  |0.5322|±  |0.0150|
| - math            |      1|custom-extract|     0|exact_match|↑  |0.8438|±  |0.0099|
| - other           |      1|custom-extract|     0|exact_match|↑  |0.7089|±  |0.0150|
| - philosophy      |      1|custom-extract|     0|exact_match|↑  |0.6553|±  |0.0213|
| - physics         |      1|custom-extract|     0|exact_match|↑  |0.8052|±  |0.0110|
| - psychology      |      1|custom-extract|     0|exact_match|↑  |0.7907|±  |0.0144|

| Groups |Version|    Filter    |n-shot|  Metric   |   |Value |   |Stderr|
|--------|------:|--------------|------|-----------|---|-----:|---|-----:|
|mmlu_pro|      2|custom-extract|      |exact_match|↑  |0.7513|±  |0.0038|
```

File: python/sglang/srt/layers/attention/flashattention_backend.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import numpy as np
 import torch
 
+from python.sglang.srt.layers.attention.triton_backend import ForwardMetadata
```
Comment (hebiao064) [inline python/sglang/srt/layers/attention/flashattention_backend.py]: do we need this?

Comment (CatherineSue) [reply python/sglang/srt/layers/attention/flashattention_backend.py]: added by accident. removed.

**Review by zhyncs — APPROVED** [review]

File: python/sglang/srt/layers/attention/flashattention_backend.py [hunk 2]
```diff
@@ -346,6 +356,8 @@ def init_forward_metadata(self, forward_batch: ForwardBatch):
                 metadata.page_table = forward_batch.req_to_token_pool.req_to_token[
                     forward_batch.req_pool_indices, : metadata.max_seq_len_k
                 ]
+
+                self._init_local_attn_metadata(metadata, device)
```
Comment (qingquansong) [inline python/sglang/srt/layers/attention/flashattention_backend.py:360]: Hey @CatherineSue for this line, have we tested this for speculative decoding topk = 1 case? Or maybe we don't have the llama eagle model yet for testing. Thanks!

Comment (CatherineSue) [reply python/sglang/srt/layers/attention/flashattention_backend.py:360]: We don't have the eagle model for testing yet. Is there any concern with topk=1?
